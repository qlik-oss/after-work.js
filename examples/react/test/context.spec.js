import React from 'react';
import renderer from 'react-test-renderer';

describe('context', () => {
  let sandbox;
  let action;
  let MyConsumerComponent;
  let hello;

  before(() => {
    sandbox = sinon.createSandbox();
    action = sandbox.spy();
    hello = 'Custom Consumer Test FOO';
    [{ default: MyConsumerComponent }] = aw.mock(
      [[require.resolve('../src/context.js'), () => props => props.children({ hello, action })]],
      ['../src/consumer'],
    );
  });

  it('renders custom consumer', () => {
    const component = renderer.create(<MyConsumerComponent />);
    const { root } = component;
    const button = root.findByType('button');
    button.props.onClick();

    expect(component.toJSON()).toMatchSnapshot();
    expect(action.called).to.equal(true);
    expect(action).calledWith(hello);
  });
});
