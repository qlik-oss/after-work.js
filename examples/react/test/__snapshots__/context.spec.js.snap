// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`'renders custom consumer' 1`] = `
<button
  onClick={[Function]}
  type="submit"
>
  Custom Consumer Test FOO
</button>
`;
